test/LiquidityHub.ts

```diff
@@ -972,4 +1034,428 @@ describe("LiquidityHub", function () {
+
+      const balanceBeforeUpgrade = await liquidityHub.balanceOf(user);
+      const totalSupplyBeforeUpgrade = await liquidityHub.totalSupply();
+      const assetBeforeUpgrade = await liquidityHub.asset();
+      const totalAssetsBeforeUpgrade = await liquidityHub.totalAssets();
+      const assetsLimitBeforeUpgrade = await liquidityHub.assetsLimit();
+
+      const liquidityHubImplV2 = (
+        await deployX("LiquidityHub", admin, "LiquidityHubV2", {}, lpToken, liquidityPool)
+      ) as LiquidityHub;
+      await liquidityHubAdmin.connect(admin).upgradeAndCall(liquidityHubProxy, liquidityHubImplV2, "0x");
+
+      expect(await liquidityHub.balanceOf(user)).to.equal(balanceBeforeUpgrade);
+      expect(await liquidityHub.totalSupply()).to.equal(totalSupplyBeforeUpgrade);
+      expect(await liquidityHub.asset()).to.equal(assetBeforeUpgrade);
+      expect(await liquidityHub.totalAssets()).to.equal(totalAssetsBeforeUpgrade);
+      expect(await liquidityHub.assetsLimit()).to.equal(assetsLimitBeforeUpgrade);
+    });
+  });
+
+  describe("ProxyAdmin Permissions", function () {
+    it("Should only allow admin to upgrade", async function () {
+      const {
+        liquidityHub, liquidityHubProxy, liquidityHubAdmin, admin, user2,
+        lpToken, liquidityPool
+      } = await loadFixture(deployAll);
+
+      const liquidityHubImplV2 = (
+        await deployX("LiquidityHub", admin, "LiquidityHubV2", {}, lpToken, liquidityPool)
+      ) as LiquidityHub;
+
+      await expect(
+        liquidityHubAdmin.connect(user2).upgradeAndCall(liquidityHubProxy, liquidityHubImplV2, "0x")
+      ).to.be.revertedWithCustomError(liquidityHubAdmin, "OwnableUnauthorizedAccount")
+        .withArgs(user2.address);
+
+      const sharesTokenBefore = await liquidityHub.SHARES();
+
+      await liquidityHubAdmin.connect(admin).upgradeAndCall(liquidityHubProxy, liquidityHubImplV2, "0x");
+
+      const sharesTokenAfter = await liquidityHub.SHARES();
+      expect(sharesTokenAfter).to.equal(sharesTokenBefore);
+    });
+
+    it("Should only allow admin to change ProxyAdmin owner", async function () {
+      const {liquidityHubAdmin, admin, user, user2} = await loadFixture(deployAll);
+
+      expect(await liquidityHubAdmin.owner()).to.equal(admin.address);
+
+      await expect(
+        liquidityHubAdmin.connect(user2).transferOwnership(user2.address)
+      ).to.be.revertedWithCustomError(liquidityHubAdmin, "OwnableUnauthorizedAccount")
+        .withArgs(user2.address);
+
+      await expect(
+        liquidityHubAdmin.connect(user).transferOwnership(user.address)
+      ).to.be.revertedWithCustomError(liquidityHubAdmin, "OwnableUnauthorizedAccount")
+        .withArgs(user.address);
+
+      await liquidityHubAdmin.connect(admin).transferOwnership(user.address);
+      expect(await liquidityHubAdmin.owner()).to.equal(user.address);
+    });
+
+    it("Should prevent direct calls to implementation contract", async function () {
```

Comment:
```suggestion
    it("Should prevent initialization of implementation contract", async function () {
```

Reply:
Also it seems the same as "Should prevent initialization on standalone implementation"
Maybe this one should just call some other functions instead of initialize?